src/Stratis.Bitcoin.Features.Apps/AppsFeature.cs

```diff
@@ -56,12 +61,23 @@ public static IFullNodeBuilder UseApps(this IFullNodeBuilder fullNodeBuilder)
                     .AddFeature<AppsFeature>()
                     .FeatureServices(services =>
                     {
+                        TryEnsureAppsFolderExists(services);
+
                         services.AddSingleton<IAppsStore, AppsStore>();                        
                         services.AddSingleton<IAppsHost, AppsHost>();
                         services.AddSingleton<AppsController>();
                     });
             });
 
+            void TryEnsureAppsFolderExists(IServiceCollection services)
+            {
+                ServiceDescriptor descriptor = services.FirstOrDefault(x => x.ServiceType == typeof(DataFolder));
```

Comment:
You dont need to query service just put DataFolder in the feature constructor and you will get it.